web/src/tree/Tree.tsx

```diff
@@ -243,12 +243,10 @@ export class Tree extends React.PureComponent<Props, State> {
                 .subscribe((props: Props) => {
                     const newParentPath = props.activePathIsDir ? props.activePath : dirname(props.activePath)
                     const queryParams = new URLSearchParams(this.props.history.location.search)
-                    // If we're updating due to a file or directory suggestion, load the relevant partial tree and jump to the file.
+                    // If we're updating due to a file/directory suggestion or code intel action,
+                    // load the relevant partial tree and jump to the file.
                     // This case is only used when going from an ancestor to a child file/directory, or equal.
-                    if (
-                        (queryParams.has('suggestion') || queryParams.has('action')) &&
-                        dotPathAsUndefined(newParentPath)
-                    ) {
+                    if (queryParams.has('subtree') && dotPathAsUndefined(newParentPath)) {
```

Comment:
Not that anything else is produced anywhere, but enforces consistency and it might confuse some poor dev that `subtree=false` does the same as `subtree=true`
```suggestion
                    if (queryParams.get('subtree') === 'true' && dotPathAsUndefined(newParentPath)) {
```